lib/common/issues/src/issue.rs

```diff
@@ -0,0 +1,66 @@
+use std::hash::{Hash, Hasher};
+
+use serde::ser::SerializeMap;
+use serde::{Serialize, Serializer};
+
+use crate::solution::Solution;
+
+/// An issue that can be hashed by its code
+pub trait Issue: Sync + Send {
+    fn code(&self) -> String;
+    fn description(&self) -> String;
+    fn solution(&self) -> Result<Solution, String>;
```

Comment:
It looks like the `solution` field uses `Result` as a generic either type, without any bias on the side.
It is a bit confusing.
Maybe we need an Enum instead?

Reply:
Agreed, that would be better.

```rust
enum Solvable {
    Yes(Solution),
    No(String),
}
```

or something like that.

Reply:
I like making it an enum instead. The `Err` variant was for when a solution couldn't be constructed for some reason, like when parsing a url. But in that case I thing the better thing to do is log and return `Solution::None`